src/include/function/unary_function_executor.h

```diff
@@ -43,6 +43,15 @@ struct UnaryStringFunctionWrapper {
     }
 };
 
+struct UnaryStructFunctionWrapper {
```

Comment:
I would try to reuse existing wrappers if possible. Maybe reuse UnaryNestedTypeFunctionWrapper?

Also, as I understand, you probably can just rewrite as literal at compile time. Evaluating at runtime seems unnecessary.